parser/parser.go

```diff
@@ -47,7 +40,15 @@ func (p *parserWrapper) Parse(ctx context.Context, file core.File) (core.ParseTr
 		return nil, fmt.Errorf("failed to read file: %w", err)
 	}
 
-	tree, err := p.parser.ParseCtx(ctx, nil, data)
+	language, exists := lang.ResolveLanguageFromPath(file.Name())
+	if !exists {
+		return nil, fmt.Errorf("failed to resolve language from file path")
+	}
+
+	parser := sitter.NewParser()
```

Comment:
AFAIK creating a new Parser for every single file will slow down the process significantly. Tree Sitter parser internally allocates memory for storing parser stage. This memory is reused while parsing multiple files. If we create 1 parser per file, it will add a lot of memory allocation & free overhead and lead to significantly high GC runs in Go.

General best practice is to have a single instance of `sitter.Parser` and use it parse and create 1 tree per file. Also the Parser needs to be closed after use so that sitter can free up internal state.

Remember: TreeSitter is written in C. We are using Go binding for TreeSitter. So we much explicitly close the state to prevent memory leak

Reply:
can we initialise a Parser with a pool of parsers beforehand ?
Would that be overload if user has asked for 10 languages .. 🤔

Reply:
@OmkarPh Good idea. We will always have a small number of Languages. We can always initialize a map in `parserWrapper` with a pool of parser keyed by `LangCode` or something. 

Take the list of languages to support for parsing as input. Fail if no language is specified. Fail if any unsupported language is specified (not sure if this is possible). This ensures client code has control over the parser and it does not do something unexpected or non-desired.

The wrapper can then resolve the language from file, lookup a parser from pool or fail if none found. Failing is important in libraries so that client code knows about the failure. Remember, decisions are in client code. We are just making it easy to perform domain specific operations like parsing, file enumeration etc.